core/dbt/adapters/base/impl.py

```diff
@@ -1009,29 +1012,56 @@ def _catalog_filter_table(
         """
         return table.where(_catalog_filter_schemas(manifest))
 
-    def _get_catalog_information_schemas(
-        self, manifest: Manifest
-    ) -> List[InformationSchema]:
-        return list(self._get_cache_schemas(manifest).keys())
+    def _get_one_catalog(
+        self,
+        information_schema: InformationSchema,
+        schemas: Set[str],
+        manifest: Manifest,
+    ) -> agate.Table:
 
-    def get_catalog(self, manifest: Manifest) -> agate.Table:
-        """Get the catalog for this manifest by running the get catalog macro.
-        Returns an agate.Table of catalog information.
-        """
-        information_schemas = self._get_catalog_information_schemas(manifest)
-        # make it a list so macros can index into it.
-        kwargs = {'information_schemas': information_schemas}
-        table = self.execute_macro(
-            GET_CATALOG_MACRO_NAME,
-            kwargs=kwargs,
-            release=True,
-            # pass in the full manifest so we get any local project overrides
-            manifest=manifest,
-        )
+        name = '.'.join([
+            str(information_schema.database),
+            'information_schema'
+        ])
+
+        # calculate the possible schemas for a given schema name
+        all_schema_names: Set[str] = set()
+        for schema in schemas:
+            all_schema_names.update({schema, schema.lower(), schema.upper()})
```

Comment:
It looks to me like we're doing case-insensitive schema filtering in the catalog queries for every plugin -- do we need to pass in a set of the lower/upper cased variants of the schema name here? In dev, I see catalog SQL like this (pg):

```
    where (upper(sch.nspname) = upper('SNAPSHOTS') or upper(sch.nspname) = upper('TEST_SCHEMA') or upper(sch.nspname) = upper('snapshots') or upper(sch.nspname) = upper('test_schema'))
```

Reply:
Ooh, you're right - this is why the comparison was case-sensitive in catalog.sql before. I'll remove this.

Reply:
that makes a ton of sense - i didn't put those two things together!